### PR TITLE
Add the CRF model wrapper

### PR DESCRIPTION
**Pull request by howl-anderson**
# Description

Brief Description of the PR:

Add the CRF model wrapper.

It can be used like following:

```python
tag_num = ... # number of tags
base_model = build_base_model(...)
model = CRFModelWraper(base_model, tag_num)

# CRFModelWraper is a subclass of tf.keras.Model
# so we can compile / fit / evaluate as normal model

# NOTE: CRF loss is computed inside the model, so we do not specific the loss for crf
model.compile(...)
model.fit(...)
```

Fixes # (issue)

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [ ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

**Comment by howl-anderson** [comment]
@gabrieldemarmiesse You are one of the most suitable reviewers for this PR. So please take a look. Thank you!

**Comment by bhack** [comment]
> @gabrieldemarmiesse You are one of the most suitable reviewers for this PR. So please take a look. Thank you!

I don't think he is still active in the project.

**Comment by howl-anderson** [comment]
@bhack Thank you for your kindly remind. Do you have any reviewer to recommend for this PR? Or could it possible to grant write permission to me? So, I can merge it to the repo. There are a few CRF related PRs still waiting for review for more than two weeks but no one response yet. If I have the write permission, I can merge them too. Of course, I am the codeower of those CRF related features, so I will continuously bugfix and improve them.

/cc @seanpmorgan

**Review by fsx950223 — APPROVED** [review]
